File: app/src/main/java/com/example/islandlyrics/WhitelistItem.java
package com.example.islandlyrics;

import java.util.Objects;

public class WhitelistItem implements Comparable<WhitelistItem> {
    private String packageName;
    private boolean isEnabled;

    public WhitelistItem(String packageName, boolean isEnabled) {
        this.packageName = packageName;
        this.isEnabled = isEnabled;
    }

    public String getPackageName() {
        return packageName;
    }

    public void setPackageName(String packageName) {
        this.packageName = packageName;
    }

    public boolean isEnabled() {
        return isEnabled;
    }

    public void setEnabled(boolean enabled) {
        isEnabled = enabled;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        WhitelistItem that = (WhitelistItem) o;
        return Objects.equals(packageName, that.packageName);
    }

    @Override
    public int hashCode() {
        return Objects.hash(packageName);
    }

    @Override
    public int compareTo(WhitelistItem other) {
        return this.packageName.compareTo(other.packageName);
    }
}

File: app/src/main/java/com/example/islandlyrics/WhitelistItem.kt
package com.example.islandlyrics

import java.util.Objects

data class WhitelistItem(var packageName: String, var isEnabled: Boolean) : Comparable<WhitelistItem> {

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null || javaClass != other.javaClass) return false
        val that = other as WhitelistItem
        return packageName == that.packageName
    }

    override fun hashCode(): Int {
        return Objects.hash(packageName)
    }

    override fun compareTo(other: WhitelistItem): Int {
        return packageName.compareTo(other.packageName)
    }
}
